lib/grid_search.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+"""
+A module that provides functionalities for grid search
+will be used for hyperparameter optimization
+"""
+
+from lib import collaborative_filtering as cf
+from util import data_parser as dp
+from lib import evaluator as ev
+import numpy
+import itertools as it
+import sys
+import os
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
```

Comment:
I'm not comfortable having this global statement. it should be in the main method.